### DISCUSSION TITLE
HK3 instead of HK2 and HK Pumpe always OFF

### DISCUSSION THREAD

**Discussion by geri167** [GyroGearl00se/ESPHome-Froeling-Lambdatronic_3200#41] [General]
Hi guys, i have the problem that my second HK is HK3 instead of HK2, HK1 is normaly and is perfect. 
So i have to modifying it but i did not work 
![image](https://github.com/user-attachments/assets/eeeb3dc2-dcba-4627-819f-98b3a340e95d)

also HK Pumpe 1 und 2 or s´3 are always OFF, why?? 

![image](https://github.com/user-attachments/assets/b438d57d-18ee-4640-a70f-b6a6a03a68e7)

**Comment by GyroGearl00se** [comment]
Hi @geri167 ,

recently I've unplugged my ESP32 as I'm currently developing also a Home Assistant Integration which will just require a "TCP to Serial Adapter" (ETH <-> RS485, RS232).

Therefore I can't test right now with this solution.

Unfortunately for me it is also HK2 ( HK1 was skipped for some reason ?).

Usually there is a offset with input or holding registers. I assume I tried to adapt this logic as well for the coils.

Could you try for the Pumpe AN/AUS following register:

HK 1 = 1030
HK 2 = 1060
HK 3 = 1090


Regarding your HK3. At first glance - the register 1090, 1091 for Vorlauf Ist- & Solltemperatur seem correct.

But the value 127 is also what I get for all unused HK's.

![image](https://github.com/user-attachments/assets/cb2480ff-7f25-4059-86c4-c5f26f554d3f)

In my case here I don't have HK1 - but I do have HK2.

The value divided by 2 (multiply 0.5) gives you the desired value in °C. Therefore 254 / = 127.


Can you confirm with the Fröling Connect App or Website that your HK is actually "3" ? The number is also displayed on the menu icon.

![image](https://github.com/user-attachments/assets/c49e3d69-13ad-4ec2-a8ec-6316a0ae7c31)

**Reply by geri167** [reply]
hy gyro, yes it is 01 and 03 

![image](https://github.com/user-attachments/assets/06702a02-4955-4cce-9d98-a0d0f4af7d84)

![image](https://github.com/user-attachments/assets/832bb577-8a6a-412f-a311-4b6620b83416)

I have a additional external control for HK03 because auf the Boiler Pumpe i think, i did not remind precisely.
